Device/Device.cs

```diff
@@ -94,9 +106,9 @@ private async void Disconnected(MqttClientDisconnectedEventArgs e, IMqttClientOp
                 Console.WriteLine("Trying to reconnect");
                 await mqttClient.ConnectAsync(options, CancellationToken.None);
             }
-            catch
+            catch (Exception ex)

```

Comment:
Is it `Exception` that is thrown in this case or is it `SomeOtherSpecificException`?

Reply:
It shoud be MqttCommunicationTimedOutException, but I did not test it myself.
https://github.com/chkr1011/MQTTnet/issues/249. There were some issues about the specific exception so I can try to test it.

Reply:
Hassaan is right, let's catch the specific exception

Reply:
But, I remove this function in my tests, because having a disconnected handler fails my test where I have to disconnect, so I'm not sure it's worth it to make these changes :D

Reply:
The method is removed in another branch.